contracts/SpeedMarkets/SpeedMarketsAMM.sol

```diff
@@ -427,10 +502,30 @@ contract SpeedMarketsAMM is Initializable, ProxyOwned, ProxyPausable, ProxyReent
             markets[i].finalPrice = market.finalPrice();
             markets[i].result = market.result();
             markets[i].isUserWinner = market.isUserWinner();
+            if (marketHasCreatedAtAttribute[marketsArray[i]]) {
+                markets[i].createdAt = market.createdAt();
+            }
         }
         return markets;
     }
 
+    /// @notice return all risk data (direction, current and max) for both directions (Up and Down) by specified asset
+    function getDirectionalRiskPerAsset(bytes32 asset) external view returns (Risk[] memory) {
+        Risk[] memory risks = new Risk[](2); // two directions: Up and Down
+        // Up
+        SpeedMarket.Direction currentDirection = SpeedMarket.Direction.Up;
+        risks[0].direction = currentDirection;
+        risks[0].current = currentRiskPerAssetAndDirection[asset][currentDirection];
+        risks[0].max = maxRiskPerAssetAndDirection[asset][currentDirection];
+        // Down
+        currentDirection = SpeedMarket.Direction.Down;
+        risks[1].direction = currentDirection;
+        risks[1].current = currentRiskPerAssetAndDirection[asset][currentDirection];
+        risks[1].max = maxRiskPerAssetAndDirection[asset][currentDirection];
+
+        return risks;
```

Comment:
maybe to use something like this for better readability 

```
SpeedMarket.Direction[] memory directions = new SpeedMarket.Direction[](2);
    directions[0] = SpeedMarket.Direction.Up;
    directions[1] = SpeedMarket.Direction.Down;

    for (uint i = 0; i < directions.length; i++) {
        SpeedMarket.Direction currentDirection = directions[i];
        risks[i].direction = currentDirection;
        risks[i].current = currentRiskPerAssetAndDirection[asset][currentDirection];
        risks[i].max = maxRiskPerAssetAndDirection[asset][currentDirection];
    }
```

Reply:
Personally I like Lotina's optimized code, even though less readable

Reply:
Will apply, it's better readability